certbot-apache/certbot_apache/_internal/configurator.py

```diff
@@ -595,6 +595,11 @@ def _deploy_cert(self, vhost, cert_path, key_path, chain_path, fullchain_path):
         # cert_key... can all be parsed appropriately
         self.prepare_server_https("443")
 
+        # If we haven't managed to enable mod_ssl by this point, error out
+        if "ssl_module" not in self.parser.modules or not self.parser.modules["ssl_module"]:
```

Comment:
One other thing...I think we should just make this line:
```suggestion
        if "ssl_module" not in self.parser.modules:
```
While we also check `not self.parser.modules["ssl_module"]` in `openssl_version()`, I think we have to do that because we need its value. Additionally, this check failing was previously only a warning and now it's a fatal error.

`"ssl_module" not in self.parser.modules` was previously a fatal error when looking for our own dummy SSL directives because of [exclude_dirs](https://github.com/certbot/certbot/blob/bcf33c665932160c2fe446e46e42c15b57c05a81/certbot-apache/certbot_apache/_internal/parser.py#L621) used in `parser.find_dir`.


Due to the way functions like [update_modules](https://github.com/certbot/certbot/blob/bcf33c665932160c2fe446e46e42c15b57c05a81/certbot-apache/certbot_apache/_internal/parser.py#L316) work, I worry that on some systems, we may not have a path here. I'm not certain there is a problem and I may be being a little paranoid, but I think it'd be safer to just check `"ssl_module" not in self.parser.modules` and I don't see any real downsides to us doing so.

If you make this change, you can also probably simplify/remove some of the other changes in this PR if you want.